src/Controls/src/Core/Platform/ModalNavigationManager/ModalNavigationManager.iOS.cs

```diff
@@ -63,8 +63,6 @@ public Task PushModalAsync(Page modal, bool animated)
 					.ModalPresentationStyle()
 					.ToNativeModalPresentationStyle();
 
-			_window.Page?.GetCurrentPage()?.SendDisappearing();
```

Comment:
The thing i see here is that now these events will fire before they actually happen on the native . Let's say i m clearing something on disappearing, could it happen that the page is still on the screen because the platform code didn't happen yet? Like it didn't awaited DismissViewControllerAsync

Reply:
The events are now firing from the call to `_navmodel.PushModal` which is directly below this call so it's now firing the same as it was for XF.  If you are wanting to clear something then you should use 'NavigatedFrom` not `Disappearing`. `NavigatedFrom` fires after the current view is now no longer visible. `Disappearing` just expresses the intent of something about to happen. 

The problem with XF also is that `Disappearing/Appearing` fired at different points in time for each platform. If you look at the Android implementation `Appearing` is firing after the animation has finished but if you look at iOS it's fired before the Modal is presented.  If you look at XF Appearing/Disappearing in non-modal apps that behavior is inverted. On iOS `Appearing` fires inside 'Didappear' and Android it fires before it appears. 

With .NET MAUI `Appearing/Disappearing` fire once the intent for those to happen are realized and they also fire for cases like backgrounding (this is all mostly the same as XF but now it's consistent)

If you want to do things after something has disppeared or once it's appeared then that's what "navigating/navigatedfrom/navigatedto" are for